File: tests/fixtures/php/t108_violation_sleep.php
<?php

use PHPUnit\Framework\TestCase;

class SleepTest extends TestCase
{
    public function test_wait_for_result(): void
    {
        startTask();
        sleep(2);
        $this->assertEquals("done", getResult());
    }

    public function test_usleep_wait(): void
    {
        usleep(500000);
        $this->assertTrue(true);
    }
}

File: tests/fixtures/rust/t108_violation_sleep.rs
use std::thread;
use std::time::Duration;

#[test]
fn test_wait_for_result() {
    start_task();
    thread::sleep(Duration::from_secs(2));
    assert_eq!(get_result(), "done");
}

#[tokio::test]
async fn test_async_wait() {
    tokio::time::sleep(Duration::from_millis(100)).await;
    assert!(true);
}
